### ISSUE TITLE
integrate cannot solve integral, but Meijer int can

### ISSUE THREAD

**Issue by Upabjojr** [sympy/sympy#11742]
Definite form can be solved by `meijerint_definite`. Apparently not linked from `integrate`:

```python
In [5]: integrate(sqrt(-x**2 + 8*x + 48), x)
Out[5]: 
⌠                        
⎮    _________________   
⎮   ╱    2               
⎮ ╲╱  - x  + 8⋅x + 48  dx
⌡                        

In [6]: meijerint.meijerint_definite(sqrt(-x**2 + 8*x + 48), x, 4, 12)
Out[6]: (16⋅π, True)
```

16⋅π is the correct result.

**Comment by Lakshya1605** [comment]
I would like to work on this issue can you please elaborate on what is expected .Thank you

**Comment by Upabjojr** [comment]
> can you please elaborate on what is expected

Well, I corrected the example. The definite integral should return 16⋅π, but it doesn't.

There is a SymPy integration algorithm that is able to calculate it (`meijerint.meijerint_definite`).

I have currently no idea on whether this is a simple or hard issue to solve.

**Comment by jksuom** [comment]
[This test](https://github.com/sympy/sympy/blob/master/sympy/integrals/integrals.py#L465) will not allow the use of `meijerint_definite` unless at least one of the limits is infinite. This restriction should probably be removed.

**Comment by Lakshya1605** [comment]
@jksuom @Upabjojr After i removed the line "(xab[1].has(oo, -oo) or xab[2].has(oo, -oo))" then the test of "test_integral.py" is running forever. Though i have changed nothing in "test_integral,py". I am not understanding what is wrong .

**Comment by Upabjojr** [comment]
@Lakshya1605 I usually run the tests in debug mode to find the reason problems occur.

**Comment by Lakshya1605** [comment]
@jksuom @Upabjojr after  modifying  code as mentioned above i am not sure why 
"def test_limit_bug():
    z = Symbol('z', zero=False)
    assert integrate(sin(x*y*z), (x, 0, pi), (y, 0, pi)) == (log(z**2) + 2*EulerGamma + 2*log(pi))/(2*z) - \ 
        (-log(pi*z) + log(pi**2*z**2)/2 + Ci(pi**2*z))/z + log(pi)/z)"
Takes infinite time when i run "./bin/test " on "test_integrals.py", whereas it gives "False" in my local machine when i run 
"integrate(sin(x*y*z), (x, 0, pi), (y, 0, pi)) == (log(z**2) + 2*EulerGamma + 2*log(pi))/(2*z) - \ 
        (-log(pi*z) + log(pi**2*z**2)/2 + Ci(pi**2*z))/z + log(pi)/z"

**Comment by jksuom** [comment]
There seems to be no problem when integrating over `x`.
```
>>> f = integrate(sin(x*y*z), (x, 0, pi), conds='none')
>>> f
-cos(pi*y*z)/(y*z) + 1/(y*z)
# or
>>> meijerint_definite(sin(x*y*z), x, 0, pi)[0].simplify()
(-cos(pi*y*z) + 1)/(y*z)
```
Also, the antiderivative of `f` wrt `y` is can be found.
```
>>> g = integrate(f, y)
>>> g
-(-log(y*z) + log(y**2*z**2)/2 + Ci(pi*y*z))/z + log(y)/z
```
However, `g` cannot be directly evaluated at `y = 0` but the limit has to be taken.
```
>>> g.subs(y, 0)
nan
>>> limit(g, y, 0)
-(log(z**2) + 2*EulerGamma + 2*log(pi))/(2*z)
```
This is the weak point of `meijerint_definite()`. The main integration function will take the limit if the substitution does not work, but `meijerint_definite()` apparently cannot do that.

The integrand `f` is already formally singular at `y = 0`.
```
>>> f.subs(y, 0)
zoo/z
```
So, a possible fix could be to allow the use of `meijerint_definite()` only when the integrand is well defined at the limit points.

**Comment by Lakshya1605** [comment]
@jksuom is their any inbuild function to check the above or should i substitute the limits in the integrand and check that none of them  goes to infinity  and -infinity?

**Comment by jksuom** [comment]
I would start by substituting a limit and scanning the result for infinities and nan, though it is not guaranteed that this would solve the issue. That can be only found out by experimenting. It is possible that something else has to be modified as well.

**Comment by Lakshya1605** [comment]
@jksuom  Still facing the same problem  not able to understand why this is happening

**Comment by Lakshya1605** [comment]
@jksuom  @Upabjojr will it be correct if i just allow the use of meijerint_definite() only when the integrand is well defined at the limit points and when integration is based only on single variable("ie . len(self.limits) == 1")

**Comment by jksuom** [comment]
`meijerint_definite()` is always concerned with integration over a single variable only. Hence it seems to me that the number of limits in the original integral is irrelevant, and should not be taken into account, only that the (possibly intermediate) integrand is well defined.

Ideally, `meijerint_definite()` should be fixed to not fall into infinite recursion (as it seems to be doing) under any conditions. This should be investigated more closely.

**Comment by Lakshya1605** [comment]
@jksuom  Can  please tell me how to "integrate(1/(1 + x + y + z), (x, 0, 1), (y, 0, 1), (z, 0, 1))" using  "meijerint.meijerint_definite " if i directly try "meijerint.meijerint_definite(1/(1 + x + y + z), (x, 0, 1), (y, 0, 1) ,( z, 0 ,1))" i get an error " unsupported operand type(s) for -: 'tuple' and 'tuple'" and if try step by step that is first integrating with respect to x then y then z then it is returning "None". Can you please help me out.Thank You.

**Comment by jksuom** [comment]
The docstring of `meijerint_definite(f, x, a, b)` says:
> Integrate ``f`` over the interval [``a``, ``b``], by rewriting it as a product
    of two G functions, or as a single G function.
    Return res, cond, where cond are convergence conditions.

so that `x, a, b ` are expected to be a symbol and two numbers. To get the integral `res` one has to take the first component of the return value.

Iterated integrals should be computed recursively one at a time, e.g., as follows.
```
>>> x, y, z = symbols('x y z', positive=True)  # positive to avoid polar numbers
>>> f = meijerint.meijerint_definite(1/(x + y + z), x, 0, 1)
>>> f[0]
log(1 + 1/(y + z))
>>> g = meijerint.meijerint_definite(f[0], y, 0, 1)
>>> g
>>> g is None
True
```
This shows that meijerint is unable to perform the second integration over `y`.

**Comment by Lakshya1605** [comment]
@jksuom ,Will it be a correct approach if i do something like this -> if meijerint returns None in any case, then we restart the integration but  this time we don't call the "try_meijerint" function any time during the integration

**Comment by jksuom** [comment]
I think that the code is currently doing that according to this comment:
>             # If the special meijerg code did not succeed in finding a definite
            # integral, then the code using meijerint_indefinite will not either
            # (it might find an antiderivative, but the answer is likely to be
            #  nonsensical).
            # Thus if we are requested to only use Meijer G-function methods,
            # we give up at this stage. Otherwise we just disable G-function
            # methods.

**Comment by Lakshya1605** [comment]
@jksuom , In the current code meijerint is used only if limits has "-oo" or "oo" even if the lines is removed (as suggested ) then as mentioned above for " integrate(1/(1 + x + y + z), (x, 0, 1), (y, 0, 1), (z, 0, 1))" .It first does meijerint integral with respect to x and returns "log(1 + 1/(y + z))" then next time it returns "None" so the function remains "log(1 + 1/(y + z))" and while finding anti derivative  returns error.
 So what i am doing is if meijerint returns "None" in any case I start integration method again from the beginning but this time not with the help of meijerint .
So it again finds integration with respect to x then with respect to y and then so on with respect to z.(Though this is a little time consuming but is working ).

**Comment by jksuom** [comment]
> while finding anti derivative returns error.

If this error is produced by meijerint_indefinite, then I think that is wrong. It should not be attempted once meijerint_definite has failed (and some flag should indicate that). But why would it be necessary to start from the beginning after some integrations in an iterated integral have succeeded. Would it not be possible to just continue without meijerint?

**Comment by Lakshya1605** [comment]
@jksuom I don't know why  but once if something is being integrated by meijerint then it is not being able to find anti derivative of the result which was returned by meijerint
.And yes I have a  used flag to indicate failure of meijerint 
Also I have started the process again because now meijerint dosent come into the picture and hence we are being able to find the anti derivative for each case.
By doing all of the above ,my code is working properly but I get many assertion errors when I run "./bin/test" though the answers are same but the way it is written are different and hence I get an assertion error can you please help me out with this.

**Comment by Upabjojr** [comment]
@Lakshya1605 if you have written some code and you wish to discuss your own code, it is best to open a PR. Describing the code by words without showing it is of little use.

**Comment by czgdp1807** [comment]
The result could have been obtained by,
```python
integrate(sqrt(-x**2 + 8*x + 48), (x, 4, 12), meijerg=True)
```
Can we close this issue and the associated PR? @Upabjojr @asmeurer

**Comment by asmeurer** [comment]
No. integrate should work without having to set a flag. That's the whole point of the issue.
